EIPS/eip-5089.md

```diff
@@ -0,0 +1,340 @@
+While custodial minting of Principal Tokens through the Principal Token contract is extremely useful for integrators, some protocols may find giving the Principal Token itself custody breaks their backwards compatability. 
+
+MUST emit the `Mint` event.
+
+MUST support ERC-20 `approve` / `transferFrom` on `asset` as a mint flow.
+MAY support an additional flow in which the underlying tokens are owned by the Principal Token contract before the `mint` execution, and are accounted for during `mint`.
+
+MUST revert if all of `amount` cannot be minted (due to deposit limit being reached, slippage, the user not approving enough underlying tokens to the PT contract, etc).
+
+Note that most implementations will require pre-approval of the Principal Token contract with the Principal Tokens `underlying` token.
+
+```yaml
+- name: mint
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: amount
+      type: uint256
+    - name: to
+      type: address
+
+  outputs:
+    - name: success
+      type: bool
+```
+
+#### redeem
+
+After maturity, burns exactly `amount` of Principal Tokens from `from` and sends the _same_ `amount` of underlying tokens to `to`.
+_Alberto: It sends underlying tokens according to `exchangeRate`._
+
+OPTIONAL - This method can be used to improve the usability and integration difficulty of Principal Tokens, but interfaces and other contracts MUST NOT expect fund custody to be present.
+While custodial redemption of Principal Tokens through the Principal Token contract is extremely useful for integrators, some protocols may find giving the Principal Token itself custody breaks their backwards compatability. 
+_Alberto: Optionality is evil._
+
+MUST emit the `Redeem` event.
+
+MUST support a redeem flow where the Principal Tokens are burned from `owner` directly where `owner` is `msg.sender` or `msg.sender` has ERC-20 approval over the shares of `owner`.
+MAY support an additional flow in which the shares are transferred to the Principal Token contract before the `redeem` execution, and are accounted for during `redeem`.
+
+MUST revert if all of `amount` cannot be redeemed (due to withdrawal limit being reached, slippage, the owner not having enough Principal Tokens, etc).
+
+Note that some implementations will require pre-requesting to the Principal Token before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: redeem
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: amount
+      type: uint256
+    - name: to
+      type: address
+    - name: from
+      type: address
+
+  outputs:
+    - name: assets
+      type: uint256
+```
+
+#### mature
+_Alberto: We surely don't need this method in the standard. It is up to each PT creator to decide how they mature it._
```

Comment:
`mature` was included to establish the workflow of an `exchangeRate` at maturity. 

I think if we instead move towards an extension of 4626 or similar, the `convertTo` methods would mean explicitly setting an `exchangeRate` at maturity isnt necessary as it becomes an implementation detail, and having `mature` in the standard would be unnecessary.

Reply:
Gove this way with the latest revision. In general, we should not include anything forcing a particular implementation.